### PR TITLE
[DOCS] Updating snapshot/restore pages to align with API changes

### PR DESCRIPTION
**Pull request by lockewritesdocs**
While adding new pages for the snapshot/restore APIs for #56069, it became apparent that the pages related to snapshot/restore (such as [Taking a snapshot](https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshots-take-snapshot.html)) contain information that is outdated, incorrect, or needs consolidation. This PR seeks to:

* Consolidate and streamline snapshot/restore information 
* Remove redundant information
* Add links to new snapshot/restore API pages
* Correct erroneous documentation

Specific changes:
* Streamline [Create a snapshot](https://elasticsearch_59730.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-take-snapshot.html) and moving GET + DELETE information to respective pages
* Introduce a dedicated [Delete a snapshot](https://elasticsearch_59730.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/delete-snapshots.html) page
* Restructure [Monitor snapshot and restore](https://elasticsearch_59730.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-monitor-snapshot-restore.html) page, plus create new section for **Retrieving snapshot status**

After creating a new page for the Restore snapshot API, I'll add links to that page from [Restore a snapshot](https://elasticsearch_59730.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-restore-snapshot.html).

Backports:
* 7.x #59803 
* 7.9 #59804 
* 7.8 #59885

### CODE REVIEW

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Use the <<delete-snapshot-api,Delete snapshot API>> to delete a snapshot
+from the repository:
+
+[source,console]
+-----------------------------------
+DELETE /_snapshot/my_backup/snapshot_1
+-----------------------------------
+
+When a snapshot is deleted from a repository, {es} deletes all files that are associated with the deleted
+snapshot and not used by any other snapshots. If the deleted snapshot operation is started while the snapshot is being
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:37]:
```suggestion
snapshot and not used by any other snapshots. If the delete snapshot operation is started while the snapshot is being
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 2]
```diff
@@ -1,11 +1,10 @@
 [[snapshots-monitor-snapshot-restore]]
 == Monitor snapshot and restore progress
-
 ++++
 <titleabbrev>Monitor snapshot and restore</titleabbrev>
 ++++
 
-There are several ways to monitor the progress of the snapshot and restore processes while they are running. Both
+Use the <<get-snapshot-api,Get snapshot API>> or the <<get-snapshot-status-api,Get snapshot status API>> to monitor the progress of the snapshot and restore processes while they are running. Both
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:7]: Hmm, this is a weird sentence now but it wasn't great before either. We simply don't have a good API to monitor restores yet (I'm on it though in #59233). Maybe just remove the restore bit from this sentence for now?

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 3]
```diff
@@ -20,71 +19,152 @@ PUT /_snapshot/my_backup
...
+return all snapshots that are currently available.
+
+Getting all snapshots in the repository can be costly on cloud-based repositories,
+both from a cost and performance perspective.  If the only information required is
+the snapshot names or UUIDs in the repository and the data streams and indices in each snapshot, then
+the optional boolean parameter `verbose` can be set to `false` to execute a more
+performant and cost-effective retrieval of the snapshots in the repository.
+
+NOTE: Setting `verbose` to `false` will omit all other information
+about the snapshot, such as status information and the number of
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:88]: > status information

This is not true, you get the status. You just don't get the error message(s) in case it's not `SUCCESS`.

**Review by original-brownbear — COMMENTED** [review]
Some minor points, looks good in general :)

Maybe have someone else look over the asciidoc formatting changes (I can't tell if everything's right there) unless you're 100% on those.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc (hunk 2 above)
Comment (lockewritesdocs) [reply docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:7]: Thanks for the heads up on #59233 -- I'll keep an eye on it. For now, let's amend this sentence to read:
>Use the <get-snapshot-api,Get snapshot API>> or the <<get-snapshot-status-api,Get snapshot status API>> to monitor the progress of snapshot operations.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc (hunk 3 above)
Comment (lockewritesdocs) [reply docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:88]: Thanks for pointing out this discrepancy. A suggested revision:

>NOTE: Setting `verbose` to `false` omits additional information
about the snapshot, such as metadata, start and end time, number of shards that include the snapshot, and error messages. The default value of the `verbose` parameter is `true`.

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 4]
```diff
@@ -0,0 +1,48 @@
+[[delete-snapshots]]
+== Delete a snapshot
+++++
+<titleabbrev>Delete snapshot</titleabbrev>
+++++
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:5]: I'd remove the title abbrev. Other pages in this section don't use them.
```suggestion
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+PUT /_snapshot/my_backup/snapshot_2?wait_for_completion=true
+
+PUT /_snapshot/my_backup/snapshot_3?wait_for_completion=true
+-----------------------------------
+// TESTSETUP
+
+////
+
+Use the <<delete-snapshot-api,Delete snapshot API>> to delete a snapshot
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:28]: We don't capitalize API names unless they contain a proper noun.
```suggestion
Use the <<delete-snapshot-api,delete snapshot API>> to delete a snapshot
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+from the repository:
+
+[source,console]
+-----------------------------------
+DELETE /_snapshot/my_backup/snapshot_1
+-----------------------------------
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:34]: I don't think the longer delimiters here help. I keep it to the minimum, which is four.
```suggestion
----
DELETE /_snapshot/my_backup/snapshot_1
----
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+When a snapshot is deleted from a repository, {es} deletes all files that are associated with the deleted
+snapshot and not used by any other snapshots. If the delete snapshot operation is started while the snapshot is being
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:37]: I'd break the para here. Seems like a different set of thoughts.
```suggestion
When a snapshot is deleted from a repository, {es} deletes all files associated with the
snapshot that are not in-use by other snapshots.

If the delete snapshot operation starts while the snapshot is being
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+created, the snapshot process halts and all files created as part of the snapshotting process are
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:38]:
```suggestion
created, the snapshot process halts and all files created as part of the process are
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+removed. Use the <<delete-snapshot-api,Delete snapshot API>> to cancel long running snapshot operations that were
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:39]: The second link seems redundant. I also think your `Use the...` sentence probably belongs before the proceeding one.
```suggestion
removed. Use the delete snapshot API to cancel long running snapshot operations that were
```

File: docs/reference/snapshot-restore/delete-snapshot.asciidoc [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+started by mistake.
+
+To delete multiple snapshots from a repository, separate snapshot names by commas or use wildcards:
+
+[source,console]
+-----------------------------------
+DELETE /_snapshot/my_backup/snapshot_2,snapshot_3
+DELETE /_snapshot/my_backup/snap*
+-----------------------------------
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/delete-snapshot.asciidoc:48]: Wildcards and comma-separated values aren't mutually exclusive. `snapshot*,my_other_snapshot*` is valid. However, you could interpret this section that way.

I'd reword and maybe add an example of two comma-separated wildcard expressions.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 11]
```diff
@@ -1,13 +1,15 @@
...
-
 ++++
 <titleabbrev>Monitor snapshot and restore</titleabbrev>
 ++++
 
-There are several ways to monitor the progress of the snapshot and restore processes while they are running. Both
-operations support `wait_for_completion` parameter that would block client until the operation is completed. This is
-the simplest method that can be used to get notified about operation completion.
+Use the <<get-snapshot-api,Get snapshot API>> or the
+<<get-snapshot-status-api,Get snapshot status API>> to monitor the
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:8]:
```suggestion
Use the <<get-snapshot-api,get snapshot API>> or the
<<get-snapshot-status-api,get snapshot status API>> to monitor the
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 12]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+}
+
 PUT /_snapshot/my_backup/snapshot_1?wait_for_completion=true
 -----------------------------------
 // TESTSETUP
 
 ////
 
-The snapshot operation can be also monitored by periodic calls to the snapshot info:
+Use the `_current` parameter to retrieve all currently running
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:39]:
```suggestion
Use the `_current` parameter to retrieve information about all currently running
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 13]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
 -----------------------------------
 GET /_snapshot/my_backup/snapshot_1
 -----------------------------------
 
-Please note that snapshot info operation uses the same resources and thread pool as the snapshot operation. So,
-executing a snapshot info operation while large shards are being snapshotted can cause the snapshot info operation to wait
-for available resources before returning the result. On very large shards the wait time can be significant.
+This request retrieves basic information about the snapshot, including start and end time, version of
+{es} that created the snapshot, the list of included data streams and indices, the current state of the
+snapshot and the list of failures that occurred during the snapshot.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:56]: Bulleted list?

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 14]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
 
-To get more immediate and complete information about snapshots the snapshot status command can be used instead:
+Similar to repositories, you can retrieve information about multiple snapshots in a single request, and wildcards are supported:
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:58]:
```
Similar to repositories,
```
Since you don't mention repositories earlier on this page, I would cut this.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 15]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+-----------------------------------
+
+Separate repository names with commas or use wildcards to retrieve snapshots from multiple repositories:
+
+[source,console]
+-----------------------------------
+GET /_snapshot/_all
+GET /_snapshot/my_backup,my_fs_backup
+GET /_snapshot/my*/snap*
+-----------------------------------
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:72]: See earlier comment.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 16]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+
+[source,console]
+-----------------------------------
+GET /_snapshot/my_backup/_all
 -----------------------------------
-// TEST[continued]
 
-While snapshot info method returns only basic information about the snapshot in progress, the snapshot status returns
+This request fails if some of the snapshots are unavailable. Use the boolean parameter `ignore_unavailable` to
+return all snapshots that are currently available.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:82]: An example snippet would help.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 17]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+
+Getting all snapshots in the repository can be costly on cloud-based repositories,
+both from a cost and performance perspective.  If the only information required is
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:85]:
```suggestion
both from a cost and performance perspective. If the only information required is
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 18]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+the snapshot names or UUIDs in the repository and the data streams and indices in each snapshot, then
+the optional boolean parameter `verbose` can be set to `false` to execute a more
+performant and cost-effective retrieval of the snapshots in the repository.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:88]: A snippet example would be helpful here.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 19]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+
+NOTE: Setting `verbose` to `false` omits additional information
+about the snapshot, such as metadata, start and end time, number of shards that include the snapshot, and error messages. The default value of the `verbose` parameter is `true`.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:91]: I don't think this needs to be in a note.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 20]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+
+[discrete]
+[[get-snapshot-detailed-status]]
+=== Retrieving snapshot status
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:95]: We try to avoid gerund titles if possible
```suggestion
=== Retrieve snapshot status
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 21]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+To retrieve more detailed information about snapshots, use the <<get-snapshot-status-api,Get snapshot status API>>. While snapshot request returns only basic information about the snapshot in progress, the snapshot status request returns
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:96]: Missing some articles here.
```suggestion
To retrieve more detailed information about snapshots, use the <<get-snapshot-status-api,get snapshot status API>>. While a snapshot request returns basic information about the snapshot in progress, the snapshot status request returns a
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 22]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
 complete breakdown of the current state for each shard participating in the snapshot.
 
-The restore process piggybacks on the standard recovery mechanism of the
-Elasticsearch. As a result, standard recovery monitoring services can be used
-to monitor the state of restore. When the restore operation is executed the
+[WARNING]
+====
+Using the Get snapshot status API to return any status results other than the currently running snapshots (`_current`) can be very expensive. Each request to retrieve snapshot status results in file reads from every shard in a snapshot, for each snapshot.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:101]: See earlier comment re: expensive.
```suggestion
Using the get snapshot status API to retrieve any status results other than the currently running snapshots (`_current`) can be very expensive. Each request to retrieve snapshot status results in file reads from every shard in a snapshot, for each snapshot.
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 23]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+returns detailed status information for the given snapshot, even
+if not currently running:
+
+[source,console]
+-----------------------------------
+GET /_snapshot/my_backup/snapshot_1/_status
+-----------------------------------
+
+[discrete]
+=== Monitoring restore operations
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:132]: Avoid gerund titles. This also needs an explicit anchor, like `[[monitor-restore-operations]]`.
```suggestion
=== Monitor restore operations
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 24]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
+primary shards is completed, {es} switches to the standard replication
+process that creates the required number of replicas. When all required
 replicas are created, the cluster switches to the `green` states.
 
 The cluster health operation provides only a high level status of the restore process. It's possible to get more
 detailed insight into the current state of the recovery process by using <<indices-recovery, index recovery>> and
 <<cat-recovery, cat recovery>> APIs.
 
-[float]
+[discrete]
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:147]: Needs an explicit anchor too.

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 25]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
 === Stop snapshot and restore operations
 
 The snapshot and restore framework allows running only one snapshot or one restore operation at a time. If a currently
-running snapshot was executed by mistake, or takes unusually long, it can be terminated using the snapshot delete operation.
-The snapshot delete operation checks if the deleted snapshot is currently running and if it does, the delete operation stops
+running snapshot was started by mistake, or takes unusually long, it can be stopped using the <<delete-snapshot-api,Delete snapshot API>>.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:151]:
```suggestion
running snapshot was started by mistake, or takes unusually long, it can be stopped using the <<delete-snapshot-api,delete snapshot API>>.
```

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc [hunk 26]
```diff
@@ -20,71 +22,151 @@ PUT /_snapshot/my_backup
...
-// TEST[continued]
 
 The restore operation uses the standard shard recovery mechanism. Therefore, any currently running restore operation can
-be canceled by deleting data streams and indices that are being restored. Please note that data for all deleted data streams and indices will be removed
+be canceled by deleting data streams and indices that are being restored. Data for all deleted data streams and indices will be removed
 from the cluster as a result of this operation.
 
-[float]
-=== Effect of cluster blocks on snapshot and restore 
+[discrete]
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:164]: Needs an explicit anchor.

File: docs/reference/snapshot-restore/restore-snapshot.asciidoc [hunk 27]
```diff
@@ -1,5 +1,8 @@
 [[snapshots-restore-snapshot]]
 == Restore a snapshot
+++++
+<titleabbrev>Restore snapshot</titleabbrev>
+++++
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/restore-snapshot.asciidoc:5]: I don't this is needed.
```suggestion
```

File: docs/reference/snapshot-restore/take-snapshot.asciidoc [hunk 28]
```diff
@@ -1,9 +1,15 @@
 [[snapshots-take-snapshot]]
-== Take a snapshot
+== Create a snapshot
+++++
+<titleabbrev>Create snapshot</titleabbrev>
+++++
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/take-snapshot.asciidoc:5]:
```suggestion
```

File: docs/reference/snapshot-restore/take-snapshot.asciidoc [hunk 29]
```diff
@@ -1,9 +1,15 @@
...
 
 A repository can contain multiple snapshots of the same cluster. Snapshots are identified by unique names within the
-cluster. A snapshot with the name `snapshot_1` in the repository `my_backup` can be created by executing the following
-command:
+cluster.
+
+Use the <<put-snapshot-repo-api,Put snapshot repository API>> to register or update a snapshot repository, and then use the <<create-snapshot-api,Create snapshot API>> to create a snapshot in a repository.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/take-snapshot.asciidoc:10]:
```suggestion
Use the <<put-snapshot-repo-api,put snapshot repository API>> to register or update a snapshot repository, and then use the <<create-snapshot-api,create snapshot API>> to create a snapshot in a repository.
```

File: docs/reference/snapshot-restore/take-snapshot.asciidoc [hunk 30]
```diff
@@ -47,167 +53,78 @@ PUT /_snapshot/my_backup/snapshot_2?wait_for_completion=true
...
-[source,console]
------------------------------------
-GET /_snapshot/my_backup/_current
------------------------------------
-// TEST[continued]
+Snapshot names can be automatically derived using <<date-math-index-names,date math expressions>>, similarly as when creating
+new data streams or indices. Special characters must be URI encoded.
 
-A snapshot can be deleted from the repository using the following command:
+For example, use the <<create-snapshot-api,Create snapshot API>> to create
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/take-snapshot.asciidoc:122]:
```suggestion
For example, use the <<create-snapshot-api,create snapshot API>> to create
```

**Review by jrodewig — APPROVED** [review]
LGTM overall. I left some nits and things to address but nothing I would consider blocking.

**Review by original-brownbear — APPROVED** [review]
LGTM now thanks!

File: docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc (hunk 26 above)
Comment (lockewritesdocs) [reply docs/reference/snapshot-restore/monitor-snapshot-restore.asciidoc:164]: Good catch! I added anchors for that section and the others that were without.

File: docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc [hunk 31]
```diff
@@ -88,9 +88,12 @@ Use the get snapshot status API to retrieve detailed information about snapshots
 
 If you specify both the repository name and snapshot, the request retrieves detailed status information for the given snapshot, even if not currently running.
 
-WARNING: Using this API to return any status results other than the currently running snapshots (`_current`) can be very expensive. Each request to retrieve snapshot status results in file reads from every shard in a snapshot, for each snapshot.
-+
+[WARNING]
+====
+Using this API to return any status results other than the currently running snapshots (`_current`) can be very expensive. Each request to retrieve snapshot status results in file reads from every shard in a snapshot, for each snapshot.
```
[the remark this answers is not in the document]
Comment (lockewritesdocs) [reply docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc:93]: Agreed. I consulted with Armin, and these requests can be expensive both for machine resources and cloud costs. Here's a revised rendition:
>Using the get snapshot status API to return any status results other than the currently running snapshots (`_current`) can be very expensive. Each request to retrieve snapshot status results in file reads from every shard in a snapshot, for each snapshot. Such requests are taxing to machine resources and can also incur high processing costs when running in the cloud.
